src/amd/compiler/aco_scheduler.cpp

```diff
@@ -68,45 +68,35 @@ void move_element(T& list, size_t idx, size_t before) {
     }
 }
 
-static std::pair<int16_t, int16_t> getLiveChanges(aco_ptr<Instruction>& instr)
+static RegisterDemand getLiveChanges(aco_ptr<Instruction>& instr)
 {
-   std::pair<int16_t, int16_t> changes{0, 0};
+   RegisterDemand changes;
    for (unsigned i = 0; i < instr->definitionCount(); i++) {
       Definition& def = instr->getDefinition(i);
       if (!def.isTemp() || def.isKill())
          continue;
-      if (def.regClass().type() == vgpr)
-         changes.second += def.size();
-      else
-         changes.first += def.size();
+      changes += def.getTemp();
    }
 
    for (unsigned i = 0; i < instr->operandCount(); i++) {
       Operand& op = instr->getOperand(i);
       if (!op.isTemp() || !op.isFirstKill())
          continue;
-      if (op.regClass().type() == vgpr)
-         changes.second -= op.size();
-      else
-         changes.first -= op.size();
+      changes -= op;
```

Comment:
I think that's the only use of the operand version. so, just use changes -= op.getTemp()

Reply:
Yeah, if it can never be a constant then we can skip this